File: GothicSwine/GothicSwine/NPC.cpp
#include "NPC.h"

/******************************************************************************************************
 * Author: Nicholas Nevins
 * Last Update: 5-30-24
 *
 * Class definitions to handle NPCs
 *****************************************************************************************************/

 /******************************************************************************************
  * Default constructor set to Nick
  *****************************************************************************************/
NPC::NPC() {
	setCharacterName("Nick");
}

/******************************************************************************************
  * Overloaded constructor for custom NPC name
  *****************************************************************************************/
NPC::NPC(std::string characterName) {
	setCharacterName(characterName);
}

/******************************************************************************************
  * Destructor
  *****************************************************************************************/
NPC::~NPC() {}

/******************************************************************************************
  * Setter for the apparel item. Requires a color, material, and type
  *****************************************************************************************/
void NPC::setApparel(Item item) {
	this->apparel = item;
}

/******************************************************************************************
  * Getter for the apparel item.
  *****************************************************************************************/
Item NPC::getApparel() { return this->apparel; }

/******************************************************************************************
  * Setter for the accessory item. Requires a color, material, and type
  *****************************************************************************************/
void NPC::setAccessory(Item item) {
	this->accessory = item;
}

/******************************************************************************************
  * Getter for the accessory item.
  *****************************************************************************************/
Item NPC::getAccessory() { return this->accessory; }

/******************************************************************************************
  * Setter for the apparel item. Requires a color, material, and type
  *****************************************************************************************/
void NPC::setFlair(Item item) {
	this->flair = item;
}

/******************************************************************************************
  * Getter for the apparel item. 
  *****************************************************************************************/
Item NPC::getFlair() { return this->flair; }

File: GothicSwinePython/Assets/NPC.py
import Assets.Item as Item

class NPC(object):
    def __init__(self, npcDict):
        self.apparel = Item.Item("apparel")
        self.accessory = Item.Item("accessory")
        self.flair = Item.Item("flair")
        self.hintToGive = ''
        self.characterName = npcDict.get('name')

    def setCharacterName(self, characterName):
        self.characterName = characterName

    def getCharacterName(self):
        return self.characterName
    
    def setApparel(self, apparel):
        self.apparel = apparel

    def getApparel(self):
        return self.apparel
    
    def setAccessory(self, accessory):
        self.accessory= accessory

    def getAccessory(self):
        return self.accessory
    
    def setFlair(self, flair):
        self.flair = flair

    def getFlair(self):
        return self.flair
    
    def setHintToGive(self, hintToGive):
        self.hintToGive = hintToGive

    def getHintToGive(self):
        return self.hintToGive
    
    def printCharacterData(self):
        print(self.getCharacterName()) 
        print(self.getAccessory().getWholeItem()) 
        print(self.getApparel().getWholeItem()) 
        print(self.getFlair().getWholeItem())
        print(self.getHintToGive())
        print(self.getGuiltyHintList())